pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -509,15 +588,20 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
+		if _, err = nc.kubeClient.Nodes().Update(node); err != nil {
 			glog.Errorf("Error updating node %s: %v", node.Name, err)
+			return gracePeriod, lastReadyCondition, readyCondition, err
 		} else {
-			return nil, gracePeriod, lastReadyCondition
+			nc.nodeStatusMap[node.Name] = NodeStatusData{
+				status:                   node.Status,
+				probeTimestamp:           nc.nodeStatusMap[node.Name].probeTimestamp,
+				readyTransitionTimestamp: nc.now(),
```

Comment:
I thought that it is always changed in this if, but now I see it's not the case if we already have 'ConditionUnknown' status.